pkg/remote/trans/default_server_handler.go

```diff
@@ -159,31 +159,39 @@ func (t *svrTransHandler) OnRead(ctx context.Context, conn net.Conn) (err error)
 		return err
 	}
 
-	var methodInfo serviceinfo.MethodInfo
-	if methodInfo, err = GetMethodInfo(ri, t.svcInfo); err != nil {
-		// it won't be err, because the method has been checked in decode, err check here just do defensive inspection
-		t.writeErrorReplyIfNeeded(ctx, recvMsg, conn, err, ri, true)
-		// for proxy case, need read actual remoteAddr, error print must exec after writeErrorReplyIfNeeded
-		t.OnError(ctx, err, conn)
-		return err
-	}
-	if methodInfo.OneWay() {
-		sendMsg = remote.NewMessage(nil, t.svcInfo, ri, remote.Reply, remote.Server)
+	// heartbeat processing
+	// recvMsg.MessageType would be set to remote.Heartbeat in previous Read procedure
+	// if specified codec support heartbeat
+	if recvMsg.MessageType() == remote.Heartbeat {
+		sendMsg = remote.NewMessage(nil, t.svcInfo, ri, remote.Heartbeat, remote.Server)
```

Comment:
Please add a test in default_server_handler_test.go to cover this branch

Reply:
ACK.